### PR TITLE
docs: update legal documents

### PR DESCRIPTION
**Pull request by tomaszbarwicki**
## Description

Updates due to changes in TRG 7:

- added Terms of Use in CONTRIBUTING file
 - extended NOTICE file with a notice for the CC-BY-4.0 license
 - added the CC-BY-4.0 license to the root repository as default license for non-code artifacts

fixes https://github.com/eclipse-tractusx/app-dashboard/issues/34

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Review by FaGru3n — APPROVED** [review]
LGTM

File: NOTICE.md [hunk 1]
```diff
@@ -18,12 +18,19 @@ source code repository logs.
...
+- Apache-2.0 for code
+- CC-BY-4.0 for non-code
+
+Apache-2.0:
+This program and the accompanying materials are made available under the terms of the Apache License, Version 2.0 which is available at https://www.apache.org/licenses/LICENSE-2.0.
 SPDX-License-Identifier: Apache-2.0
 
+CC-BY-4.0:
+The materials in this repository are made available under the terms of the Creative Commons Attribution 4.0 International License, which is available at https://spdx.org/licenses/CC-BY-4.0.html.
+SPDX-License-Identifier: CC-BY-4.0
```
Comment (carslen) [inline NOTICE.md:32]:
```suggestion
- Apache-2.0 for code
- CC-BY-4.0 for non-code

## Apache-2.0:

This program and the accompanying materials are made available under the terms of the Apache License, Version 2.0 which is available at https://www.apache.org/licenses/LICENSE-2.0.

SPDX-License-Identifier: Apache-2.0

### CC-BY-4.0:

The materials in this repository are made available under the terms of the Creative Commons Attribution 4.0 International License, which is available at https://spdx.org/licenses/CC-BY-4.0.html.

SPDX-License-Identifier: CC-BY-4.0
```
For better readability/Markdown rendering.

**Review by carslen — COMMENTED** [review]
The rendering will be hard to read, because the linebreaks are ignored in Markdown. So I added new Subheadings and added some more line breaks for readability in my suggestion.

**Comment by tomaszbarwicki** [comment]
Created another PR to fix Quality Checks non root container failure - https://github.com/eclipse-tractusx/app-dashboard/pull/36

**Review by SebastianBezold — APPROVED** [review]

**Pull request merged**